### PR TITLE
fix(dashboard): make to refresh all tab content charts

### PR DESCRIPTION
**Pull request by prosdev0107**
### SUMMARY
"Refresh Dashboard" only refreshes active tab
**Description**
On a tabbed dashboard, “refresh dashboard” only refreshes the active tab

**Expected results**
all tabs should refresh

**Actual results**
only the active tab is refreshed

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
BEFORE:

https://user-images.githubusercontent.com/47900232/167009037-d346720f-b8e4-4bfe-8e24-e9125ec55526.mov

AFTER:

https://user-images.githubusercontent.com/47900232/167009029-f2d4101a-7012-46a8-946f-f10e8ac79f92.mov

### TESTING INSTRUCTIONS

**How to reproduce the bug**

1. Go to a tabbed dashboard
2. Change the underlying dataset of some charts do the visualization changes
3. Click on 'refresh dashboard'
4. Only the active tab is refreshed, other tabs are not

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/gridComponents/Chart.jsx [hunk 1]
```diff
@@ -154,6 +157,10 @@ export default class Chart extends React.Component {
       return true;
     }
 
+    if (!isEqual(nextProps.chart, this.props.chart)) {
```
Comment (diegomedina248) [inline superset-frontend/src/dashboard/components/gridComponents/Chart.jsx:160]: is this different than `if (!areObjectsEqual(nextProps[prop], this.props[prop]))`?
If you don't filter the chart prop to `SHOULD_UPDATE_ON_PROP_CHANGES`, shouldn't the code reach the `areObjectEquals`?

Comment (prosdev0107) [reply superset-frontend/src/dashboard/components/gridComponents/Chart.jsx:160]: @diegomedina248 
I think that `isEqual` works similar to `areObjectsEqual` and `areObjectsEqual` maybe more strict in this case.

And then, when click `refresh dashboard` after change the dataset, all chart components should update whether if it is now visible or not. Thus, I define the filtering by chart props outside of `isComponentVisible` props filtering and so `chart` prop doesn't need to reach the `areObjectEquals` in `SHOULD_UPDATE_ON_PROP_CHANGES `.

Comment (diegomedina248) [reply superset-frontend/src/dashboard/components/gridComponents/Chart.jsx:160]: mm, but I think `isComponentVisible` might be broader, and put in place for a reason.
If we add this condition here, we might be triggering more rerenders than expected.
@rusackas what do you think?

Comment (stephenLYZ) [reply superset-frontend/src/dashboard/components/gridComponents/Chart.jsx:158]: It makes sense to not render the chart when it is not visible, which improves performance. The same is true for fine-grained control of chart updates via SHOULD_UPDATE_ON_PROP_CHANGES.

In this issue, in the case of a chart that is not updated, I suggest finding a prop that is not updated case by case and add it to SHOULD_UPDATE_ON_PROP_CHANGES.

File: superset-frontend/src/dashboard/components/gridComponents/Chart.jsx [hunk 2]
```diff
@@ -175,6 +180,11 @@ export default class Chart extends React.Component {
         this.resizeTimeout = setTimeout(this.resize, RESIZE_TIMEOUT);
       }
 
+      if (nextState.isUpdated) {
+        this.setState({ isUpdated: false });
```
Comment (stephenLYZ) [inline superset-frontend/src/dashboard/components/gridComponents/Chart.jsx:184]: It is not recommended to call `setState` in SCU, it is rather hacky. If you want to change state when a prop change, please consider using [static getDerivedStateFromProps](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops).

**Comment by mistercrunch** [comment]
Hey it's hard to review the PR since you probably `mv` the file instead of `git mv`. If you `git mv` we'll be able to diff the file in the PR as opposed to seeing the old file deleted and the new file show as new.

**Pull request closed without merging**